### PR TITLE
feat: add donation page

### PR DESCRIPTION
**Pull request by oSumAtrIX**
https://github.com/ReVanced/revanced-website/assets/13122796/4dc7cf75-3352-4232-b8d5-9d71035c9de4

# About

This PR aims to add a donation page that can be linked instead of the donation options separately everywhere.

# TODO

- [x] Resolve duplicate heart SVG and the path in the nav bar
- [x] Improve responsiveness
- [x] Proper padding of the pulsating logo
- [x] Improve semantics of the donation page text content
- [x] Add a "Copied" tooltip to the crypto keys
- [x] ~~Shorten crypto key strings to the following format: `D8G...LXp` (How to handle this on mobile?)~~ Decided against, keys arent as long as expected
- [x] Add QR codes to crypto keys
- [x] Instead of hardcoding donation links and keys, use API

### CODE REVIEW

**Comment by alexandreteles** [comment]
@Ushie Maybe fix and merge? :)

**Comment by Ushie** [comment]
I lack the frontend knowledge to carry on this PR

**Comment by Ushie** [comment]
> Add QR codes to crypto keys
@oSumAtrIX Do you have any idea how this would look like?

**Comment by Ushie** [comment]
@oSumAtrIX Is the animated image going to be included? otherwise this is ready to merge

File: src/layout/Footer/FooterHost.svelte [hunk 1]
```diff
@@ -68,18 +69,27 @@
...
 					<!-- to replace -->
 					<li><a href="https://github.com/revanced" target="_blank" rel="noreferrer">GitHub</a></li>
-					<li><a href="https://revanced.app/discord" target="_blank" rel="noreferrer">Discord</a></li>
-					<li><a href="https://reddit.com/r/revancedapp" target="_blank" rel="noreferrer">Reddit</a></li>
+					<li>
+						<a href="https://revanced.app/discord" target="_blank" rel="noreferrer">Discord</a>
+					</li>
+					<li>
+						<a href="https://reddit.com/r/revancedapp" target="_blank" rel="noreferrer">Reddit</a>
+					</li>
```
Comment (oSumAtrIX) [inline src/layout/Footer/FooterHost.svelte:77]: Links should be consumed via the API

File: static/revanced-logo-background.svg
Comment (oSumAtrIX) [inline static/revanced-logo-background.svg:1]: This file is not rendered anywhere on the page

File: static/icons/heart.svg
Comment (oSumAtrIX) [inline static/icons/heart.svg:1]: This file is not rendered anywhere on the page

File: src/routes/donate/+page.svelte [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+	<p>
+		ReVanced offers a variety of patches, including ad-blocking, custom themes, and innovative
+		features. All of which is completely open source and free of charge. Donating will allow
+		ReVanced maintain our servers and develop new features.
+	</p>
+	<Query query={donateQuery} let:data>
+		{#if data.platforms}
+			<div class="buttons-container">
+				{#each data.platforms as platform, i}
+					{#if i === 0}
```
Comment (oSumAtrIX) [inline src/routes/donate/+page.svelte:47]: The API does not return an ordered list. The API should support a "primary" method which you would use to display the button for it filled.

File: src/routes/donate/+page.svelte [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+						{#each data.wallets as wallet}
+							<button
+								on:click={() => {
+									qrCodeValue = wallet.address;
+									qrCodeDialogueName = wallet.currency_code;
+									qrCodeDialogue = !qrCodeDialogue;
+								}}
+							>
+								<div class="name">
+									<img src="/donate/{wallet.currency_code}.svg" alt={`${wallet.network} icon.'`} />
```
Comment (oSumAtrIX) [inline src/routes/donate/+page.svelte:83]: How do you handle currencies that you do not have an SVG for? You should default to a generic image.

File: src/lib/components/AnimatedImage.svelte
Comment (oSumAtrIX) [inline src/lib/components/AnimatedImage.svelte:1]: This component is not used anywhere, you could place it somewhere on the page as illustrated in the PR description.

File: src/layout/Footer/FooterHost.svelte (hunk 1 above)
Comment (Ushie) [reply src/layout/Footer/FooterHost.svelte:77]: This is out of scope of the PR

**Comment by oSumAtrIX** [comment]
@xafn Removing the border-radius from the image would look like this:

image

Before:

image

The reason for this is that it may look odd due to the high difference in contrast. Because there is a lot of light in the corners, the eye pupil closes to prevent too much light from getting in. When it closes, due to less lightness, the contrast of the card background and the actual website background is reduced resulting in a perceived rounding of the card background at the corners in a low light environment as illustrated:

https://github.com/ReVanced/revanced-website/assets/13122796/86100006-39ec-49d4-9928-8384a2c4692b

File: src/routes/donate/+page.svelte [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+						<span>{platform.name}</span>
+					</a>
+				{/each}
+			{/if}
+			{#if data.wallets}
+				<button class="donate-card" on:click={() => (cryptoDialogue = !cryptoDialogue)}>
+					<div
+						style="background-image: url('/donate/card-images/Cryptocurrency.png');"
+					/>
+					<span>Cryptocurrency</span>
```
Comment (oSumAtrIX) [inline src/routes/donate/+page.svelte:77]:
```suggestion
					<span>Cryptocurrencies</span>
```

```suggestion
					<span>Crypto</span>
```

**Pull request merged**